java/client/src/main/java/glide/connectors/handlers/ChannelHandler.java

```diff
@@ -23,6 +23,10 @@ public class ChannelHandler {
     protected final Channel channel;
     protected final CallbackDispatcher callbackDispatcher;
 
+    public boolean isClosed() {
+        return !channel.isOpen();
```

Comment:
I'm not sure that it is an immediate transition. `channel.close` is an async operation so you (or user) still may be able to write something to a channel is being closed

Reply:
And if a user writes something to the channel before the channel.close() call finishes, we could get in a state where the `channel` is still open (and will accept a request and add a request to the list - which will never complete).  
@SanHalacogluImproving let's change this a private boolean field that saves the state of the channelHandlers.

```
   private boolean isClosed = false;

    public boolean isClosed() {
        return this.isClosed;
    }

    public ChannelFuture close() {
        this.isClosed = true;
        callbackDispatcher.shutdownGracefully();
        return channel.close();
    }
```